netbsd/NetBSDProcessList.c

```diff
@@ -0,0 +1,314 @@
+/*
+htop - NetBSDProcessList.c
+(C) 2014 Hisham H. Muhammad
+(C) 2021 Santhosh Raju
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "NetBSDProcessList.h"
+
+#include <kvm.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/mount.h>
+#include <sys/param.h>
+#include <sys/proc.h>
+#include <sys/sched.h>
+#include <sys/swap.h>
+#include <sys/sysctl.h>
+#include <sys/types.h>
+#include <uvm/uvm_extern.h>
+
+#include "CRT.h"
+#include "Macros.h"
+#include "Object.h"
+#include "NetBSDProcess.h"
+#include "Process.h"
+#include "ProcessList.h"
+#include "Settings.h"
+#include "XUtils.h"
+
+static long fscale;
+static int pageSize;
+static int pageSizeKB;
+
+ProcessList* ProcessList_new(UsersTable* usersTable, Hashtable* pidMatchList, uid_t userId) {
+   const int mib[] = { CTL_HW, HW_NCPU };
+   const int fmib[] = { CTL_KERN, KERN_FSCALE };
+   int r;
+   size_t size;
+   char errbuf[_POSIX2_LINE_MAX];
+
+   NetBSDProcessList* opl = xCalloc(1, sizeof(NetBSDProcessList));
+   ProcessList* pl = (ProcessList*) opl;
+   ProcessList_init(pl, Class(NetBSDProcess), usersTable, pidMatchList, userId);
+
+   size = sizeof(pl->cpuCount);
+   r = sysctl(mib, 2, &pl->cpuCount, &size, NULL, 0);
+   if (r < 0 || pl->cpuCount < 1) {
+      pl->cpuCount = 1;
+   }
+   opl->cpus = xCalloc(pl->cpuCount + 1, sizeof(CPUData));
+
+   size = sizeof(fscale);
+   if (sysctl(fmib, 2, &fscale, &size, NULL, 0) < 0) {
+      CRT_fatalError("fscale sysctl call failed");
+   }
+
+   if ((pageSize = sysconf(_SC_PAGESIZE)) == -1)
+      CRT_fatalError("pagesize sysconf call failed");
+   pageSizeKB = pageSize / ONE_K;
+
+   for (unsigned int i = 0; i <= pl->cpuCount; i++) {
+      CPUData* d = opl->cpus + i;
+      d->totalTime = 1;
+      d->totalPeriod = 1;
+   }
+
+   opl->kd = kvm_openfiles(NULL, NULL, NULL, KVM_NO_FILES, errbuf);
+   if (opl->kd == NULL) {
+      CRT_fatalError("kvm_openfiles() failed");
+   }
+
+   return pl;
+}
+
+void ProcessList_delete(ProcessList* this) {
+   NetBSDProcessList* opl = (NetBSDProcessList*) this;
```

Comment:
Consistently this would be called `npl` instead of `opl`.

Reply:
Should be fixed in 50ae2bd